docs/research/ha/RATIONALE.md

```diff
@@ -0,0 +1,446 @@
+the easiest thing to explore and to investigate.
+
+## Failure cases
+
+The `PV` state is kept coherent by Longhorn, so errors at this level are assumed
+NOT possible (application level corruptions to the `PV`'s state ARE possible).
+
+What are the failure cases that can happen for s3gw?
+
+Let's examine the following scenarios:
+
+1. `radosgw`'s POD failure or rescheduling
+2. Cluster's node failure
+3. `radosgw`'s failure due to a bug not related to a certain input pattern
+4. `radosgw`'s failure due to a bug related to a certain input pattern
+5. `PV` Data corruption at application level due to `radosgw`'s bug
+6. `PV` Data corruption at application level due to `radosgw`'s anomalous exit
+
+We are supposing all these bugs or conditions to be fatal for the s3gw's process
+so that they trigger an anomalous exit.
+
+### radosgw's POD failure
+
+This case is when the `radosgw` process stops due to a failure of its POD.
+This case applies also when Kubernetes decides to reschedule the POD
+to another node, eg: when the node goes low on resources.
+
+This can be thought as an infrastructure issue independent to the s3gw.
+In this case, the *One Active pipeline* model fully restores the service by
+rescheduling a new POD somewhere in the cluster.
+
+### Cluster's node failure
+
+This case is when the `radosgw` process stops due to a cluster's node failure.
+The *One Active pipeline* model fully restores the service by
+rescheduling a new POD somewhere in the cluster.
+
+### radosgw's failure due to a bug not related to a certain input pattern
+
+This case is when the `radosgw` process crashes due to a bug not directly
+related to any input type.
+
+Examples:
+
+- Memory leaks
+- Memory corruptions (stack or heap corruptions)
+- Periodic operations or routines not related to an input (GC, measures,
+  telemetry, etc)
+
+For a bug like this, the *One Active pipeline* model could guarantee
+the user's operations *until the next occurrence* of the same malfunctioning.
+
+A definitive solution would be available only when a patch for the issue
+has released.
+
+### radosgw's failure due to a bug related to a certain input pattern
+
+This case is when the `radosgw` process crashes due to a bug directly
+related to a certain input type.
+
+Examples:
+
+- Putting Buckets with some name's pattern
+- Putting Objects that have a certain size
+- Performing an admin operation over a suspended user
+
+For a bug like this, the *One Active pipeline* model could guarantee
+the user's operations under the condition that the crash-triggering
+input is recognized by the user and thus its submission blocked.
+
+### PV Data corruption at application level due to radosgw's bug
+
+This case is when the state on the `PV` corrupts due to a `radosgw`
+bug.
+
+In this unfortunate scenario, the *One Active pipeline* can hardly help.
+A restart could evenly fix the problem or trigger an endless restarting loop.
```

Comment:
Right, none of the models can help here. Logical data corruption is a Robustness/Reliability problem; the best we can aim for is to detect it (and abort with prejudice and finally so, so as to not make the corruption worse).
We can't even tell it apart from data corruption at the actual PV level; all we know the data isn't consistent/coherent.

I think beyond that ("don't make it worse"), this case is outside the scope of an HA scenario. There's no way to recover here. This is more related to Tim's work on the s3gw.fsck